tests/python/gaia-ui-tests/gaiatest/form_controls/binarycontrol.py

```diff
@@ -0,0 +1,58 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from abc import ABCMeta, abstractproperty
+from marionette_driver import Wait, expected
+from gaiatest.form_controls.form_control import FormControl
+
+
+class BinaryControl(FormControl):
+    # This represent elements like Switches and Checkboxes
+
+    __metaclass__ = ABCMeta
+
+    @abstractproperty
+    def is_checked(self):
+        pass
+
+    def enable(self):
+        self._toggle_and_verify_state(final_state=True)
+
+    def disable(self):
+        self._toggle_and_verify_state(final_state=False)
+
+    def _toggle_and_verify_state(self, final_state):
+        Wait(self.marionette).until(expected.element_enabled(self.root_element))
+        Wait(self.marionette).until(lambda m: self.is_checked is not final_state)
+        self._toggle()
+        Wait(self.marionette).until(lambda m: self.is_checked is final_state)
+        Wait(self.marionette).until(expected.element_enabled(self.root_element))
```

Comment:
I guess this Wait is necessary for cases like the Wifi checkbox in settings disable, right?